controller/src/test/java/io/pravega/controller/timeout/TimeoutServiceTest.java

```diff
@@ -225,13 +225,13 @@ public void testPingSuccess() throws InterruptedException {
         PingTxnStatus pingStatus = timeoutService.pingTxn(SCOPE, STREAM, txData.getId(), txData.getVersion(), LEASE);
         Assert.assertEquals(PingTxnStatus.Status.OK, pingStatus.getStatus());
 
-        result = timeoutService.getTaskCompletionQueue().poll((long) (0.5 * LEASE), TimeUnit.MILLISECONDS);
+        result = timeoutService.getTaskCompletionQueue().poll((long) (0.1 * LEASE), TimeUnit.MILLISECONDS);
```

Comment:
we should infact do the following: 
have larger lease to guarantee that the lease has not expired and hence task queue is empty. 

and later in the code where we want the lease to take expired, we should poll for at least greater than lease period. 

but this second aspect can make this test run for long if the lease period is large.

So ideal fix would be to inject a clock in the timeoutService.. and progress the time manually from the test so that we can guarantee when the clock goes past the lease. 
otherwise we always have to choose between a large enough lease to not run into these timing issues vs choosing a small time to ensure test does not take long to complete.